easybuild/framework/easyblock.py

```diff
@@ -72,7 +72,9 @@
 from easybuild.tools.filetools import compute_checksum, copy_file, derive_alt_pypi_url, diff_files, download_file
 from easybuild.tools.filetools import encode_class_name, extract_file, is_alt_pypi_url, mkdir, move_logs, read_file
 from easybuild.tools.filetools import remove_file, rmtree2, verify_checksum, weld_paths, write_file
-from easybuild.tools.hooks import run_hook
+from easybuild.tools.hooks import BUILD_STEP, CLEANUP_STEP, CONFIGURE_STEP, EXTENSIONS_STEP, FETCH_STEP, INSTALL_STEP
```

Comment:
Not to be picky, but wouldn't it make sense to put the steps in a dict?

Reply:
How do you envision that exactly? What would be the key, what the value?

Also, dictionaries are mutable, so not a good choice for constants...

Reply:
I'd do something like:
```
steps_dict = {
    'BUILD_STEP': 'build',
    'CLEANUP_STEP': 'cleanup',
    ....
}
steps = frozenset(steps_dict.items())
```

And then import `steps` and use `dict(steps)['BUILD_STEP']` for instance. It isn't too elegant, but importing every step individually seems like a PITA, and more difficult to maintain.

Reply:
@damianam That's kind of bypassing the whole point of introducing constants.

The point here is to avoid "magic strings" like `'build'`, while typos in constants like `BUILD_STEP` are easier to catch by linters, etc.
With your proposal, we're back to magic strings like `'BUILD_STEP'`...

Also, w.r.t. to maintenance: adding additional steps should be rare in the future, removing steps even more so.

I agree that the import of the constants looks a bit ugly though, but I don't see a better option without bypassing the purpose of using constants.

Reply:
You are right. We can leave it as it is and blame it on the language ;-)